snapcraft/internal/meta/_snap_packaging.py

```diff
@@ -180,22 +176,23 @@ def _icon_file_exists() -> bool:
         return False
 
 
-def _get_app_name_from_desktop_file_id(
-        config_data: Dict[str, Any], desktop_file_id: str) -> str:
-    """Get the snap app name from a desktop file ID.
+def _get_app_name_from_desktop_file_path(
+        config_data: Dict[str, Any], desktop_file_path: str) -> str:
+    """Get the snap app name from a desktop file path.
 
-    Desktop file IDs are defined in
-    https://standards.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#desktop-file-id
-
-    XXX We try to find a match between desktop file id and snap app name.
+    XXX We try to find a match between desktop file path and snap app name.
     This will be much nicer once snapcraft supports appstream IDs:
     https://forum.snapcraft.io/t/support-for-appstream-id/2327
     --elopio - 20180108
 
-    :params desktop_file_id: The identifier of the desktop file.
+    :params desktop_file_path: The path to the desktop file.
     :returns: The name of the snap app that corresponds to the desktop file.
 
-    """  # noqa
+    """
+    desktop_file_id = desktop_file_path[
+        desktop_file_path.find('applications') +
```

Comment:
If this happens to not be in the string, you'll get a -1. I don't feel like that will produce desired results, will it?

Reply:
`_desktop_file_id_to_path` would seem to force that. But I was wondering, do we want to ignore other desktop files entirely? Currently this isn't required to just use them as launchers.

Reply:
Only if they're coming from appstream. We're making this change to make this easier to use from other metadata extraction methods (file paths instead of IDs), but this section seems to be pretty specific to appstream.

Reply:
But remember this guessing will go away when we add the appstream id. If `applications` is not in the path, I would prefer just to skip instead of try another guess. But, no strong opinion here, I can go with your option.

Reply:
Alright, that's fine then. However, can you update the code to bail if applications isn't found instead of continuing the search with a -1? That will make it more clear that we don't intend for that to work (along with a comment, perhaps).